### PR TITLE
6132 - Maven Offline support

### LINKED ISSUE #6132: Make efficient use of Maven caching in CI and Docker builds
This is related to my other dependency housekeeping things. See #5288 and #5360.
Also related to IQSS/dataverse-kubernetes#64.

----

1. Maven downloads the world for dependencies and again for plugins. These downloads should be avoided when the POM is unchanged.
2. All modern CI tools like Travis, Jenkins, GitLab, ... offer caching for reuse during next build.
3. Docker layers offer caching.

To efficiently cache all dependencies and plugins, one needs to tell Maven it shall "download the world" before executing anything else. After that, Maven can be used in offline mode: `mvn -o ...`.

However, the official dependency plugin target `mvn dependency:go-offline` does _NOT_ download everything which is necessary: plugins are missing. There are some bugs for this, see [MDEP-82](https://issues.apache.org/jira/browse/MDEP-82), [MDEP-516](https://issues.apache.org/jira/browse/MDEP-516) and many more, but this has not been resolved yet.

For further enhancement of our CI and Docker efforts, I would like to add a Maven plugin. It will do no harm to anyone else, as you need to explicitly use it. Otherwise it will do just nothing.

### PR DESCRIPTION
**Pull request by poikilotherm**
## Related Issues

- Closes #6132

## Pull Request Checklist

- [x] [Documentation][docs] completed
- [x] Merged latest from "develop" [branch][] and resolved conflicts

[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -691,26 +687,15 @@
...
-                </executions>
+            </plugin>
+            <plugin>
+                <groupId>de.qaware.maven</groupId>
+                <artifactId>go-offline-maven-plugin</artifactId>
+                <version>1.2.1</version>
+                <configuration>
+                    <dynamicDependencies>
+                    </dynamicDependencies>
+                </configuration>
```
Comment (pdurbin) [inline pom.xml]: @poikilotherm could this pull request be simplified to only be about adding the new "offline" plugin? What I'm trying to ask is if all the other changes about "endorsed" and "executions" are required for you to get what you immediately want.

Comment (poikilotherm) [reply pom.xml]: I had some troubles when I was going to solve this via the normal dependency plugin first.
It crashed due to the missing endorsed API jar, so I tried to eliminate root cause.
If you guys feel better leaving out this commit, no problem. It should be cleaned up anyway, but might be pulled into a separate issue.

Comment (pdurbin) [reply pom.xml]: @poikilotherm yes can you please remove that commit to make the pull request smaller and less controversial? You are welcome to follow up with a future pull request or two for the "endorsed" and "executions" stuff.

Comment (poikilotherm) [reply pom.xml]: Done. Will leave it there for now.

**Comment by poikilotherm** [comment]
Heads up @pdurbin, I just reverted 7bc6c6f9dd. Hope this is ready to go now.

**Comment by pdurbin** [comment]
@poikilotherm just reported some problems with the branch and we decided it isn't ready for QA: http://irclog.iq.harvard.edu/dataverse/2019-09-03#i_104215

**Comment by poikilotherm** [comment]
#6140 and its PR #6141 are blockers for this.

**Comment by poikilotherm** [comment]
As #6140 has been solved by merging #6141, this can be moved to QA again.

**Pull request merged**